### ISSUE TITLE
What are MongoManagedTypes?

### ISSUE THREAD

**Issue by emoen** [spring-projects/spring-data-mongodb#4365]
MongoManagedTypes are not mentioned in the v 4.0 release even though they where released in that version. They are needed when overriding mongoMappingContext(MongoCustomConversions customConversions, MongoManagedTypes types) which are needed for custom converters. Can you update java-doc and give an example where there is 2 or more customConverters?

**Comment by christophstrobl** [comment]
Thanks for pointing out the blank spot in the reference documentation.

As per the Javadoc of the `ManagedTypes` interface those are:
> Types managed by a Spring Data implementation. Used to predefine a set of known entities that might need processing during the initialization phase.

Basically its the initial entity set now exposed as a dedicated type which helps detect them during a potential AOT phase to run inspections required when building a GraalVM native-image.

**Comment by emoen** [comment]
Yes, I found the commit where the change was done: [java/org/springframework/data/mongodb/config/MongoConfigurationSupport.java](https://github.com/spring-projects/spring-data-mongodb/commit/cfd55be95bf7bf70bc1fdf243209591c0b0c4d5b#diff-ad4017207d4ae8d4873841039c15eb381df9a5caaf06a5e2c4b9f24e01081c69)

**Comment by emoen** [comment]
Before it was mappingContext.setInitialEntitySet(getInitialEntitySet()); and now it is mappingContext.setManagedTypes(managedTypes);
So my question is - how to I migrate with this change in mind - now that mongoMappingContext() takes a new argument as input? Related stackexchange question
https://stackoverflow.com/questions/75996137/what-are-mongomanagedtypes-new-in-spring-data-mongodb-version-4-0

**Comment by emoen** [comment]
> Basically its the initial entity set now exposed as a dedicated type which helps detect them during a potential AOT phase to run inspections required when building a GraalVM native-image.
How can I set them?

**Comment by christophstrobl** [comment]
Can you share a complete [minimal sample](https://stackoverflow.com/help/minimal-reproducible-example) (something that we can unzip or git clone, build, and deploy) that shows what you're trying to achieve. I seem to lack some context form the linked SO question and the error posted there.

**Comment by emoen** [comment]
The attached image shows the problem - in spring-data-mongodb 4.0 the method getMappingBasePackages() which is defined with @Override is not called anymore: https://ibb.co/ZdChJbQ

**Comment by christophstrobl** [comment]
Why do you think `getMappingBasePackages` is no longer called?
`MongoConfigurationSupport` uses `getInitialEntitySet` to create the `MongoManagedTypes` bean out of the configured base packages. As already said, please provide a complete [minimal sample](https://stackoverflow.com/help/minimal-reproducible-example) that reproduces the problem. Screenshots unfortunately contain way to little information to be of value.

**Comment by christophstrobl** [comment]
Then providing the sample should not be issue.
